plugin/evm/gossip_mempool.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package evm
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ava-labs/avalanchego/network/p2p/gossip"
+
+	"github.com/ava-labs/coreth/core"
+	"github.com/ava-labs/coreth/core/txpool"
+	"github.com/ava-labs/coreth/core/types"
+	"github.com/ava-labs/coreth/plugin/evm/message"
+)
+
+var (
+	_ gossip.Gossipable        = (*GossipEthTx)(nil)
+	_ gossip.Gossipable        = (*GossipAtomicTx)(nil)
+	_ gossip.Set[*GossipEthTx] = (*GossipEthTxPool)(nil)
+)
+
+type GossipAtomicTx struct {
+	Tx *Tx `serialize:"true"`
+}
+
+func (tx *GossipAtomicTx) GetID() ids.ID {
+	return tx.Tx.ID()
+}
+
+func (tx *GossipAtomicTx) Marshal() ([]byte, error) {
+	return Codec.Marshal(message.Version, tx)
+}
+
+func (tx *GossipAtomicTx) Unmarshal(bytes []byte) error {
+	_, err := Codec.Unmarshal(bytes, tx)
+	return err
+}
+
+func NewGossipEthTxPool(mempool *txpool.TxPool) (*GossipEthTxPool, error) {
+	bloom, err := gossip.NewBloomFilter(txGossipBloomMaxItems, txGossipBloomFalsePositiveRate)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize bloom filter: %w", err)
+	}
+
+	return &GossipEthTxPool{
+		mempool:    mempool,
+		pendingTxs: make(chan core.NewTxsEvent),
+		bloom:      bloom,
+	}, nil
+}
+
+type GossipEthTxPool struct {
+	mempool    *txpool.TxPool
+	pendingTxs chan core.NewTxsEvent
+
+	bloom *gossip.BloomFilter
+	lock  sync.RWMutex
+}
+
+func (g *GossipEthTxPool) Subscribe(shutdownChan chan struct{}, shutdownWg *sync.WaitGroup) {
+	defer shutdownWg.Done()
+	g.mempool.SubscribeNewTxsEvent(g.pendingTxs)
+
+	for {
+		select {
+		case <-shutdownChan:
+			log.Debug("shutting down subscription")
+			return
+		case pendingTxs := <-g.pendingTxs:
+			g.lock.Lock()
+			for _, pendingTx := range pendingTxs.Txs {
+				tx := &GossipEthTx{Tx: pendingTx}
+				g.bloom.Add(tx)
+				reset, err := gossip.ResetBloomFilterIfNeeded(g.bloom, txGossipBloomMaxFilledRatio)
+				if err != nil {
+					log.Error("failed to reset bloom filter", "err", err)
+					continue
+				}
+
+				if reset {
+					log.Debug("resetting bloom filter", "reason", "reached max filled ratio")
+
+					pending := g.mempool.Pending(false)
```

Comment:
Can we document why we pass `true` or `false` into `Pending`? We pass both values in different situations without much explanation

Reply:
We can remove this call with a call to `IteratePending` I think.